test/NormalModule.test.js

```diff
@@ -0,0 +1,410 @@
+			it("returns an empty array", function() {
+				normalModule.splitVariablesInUniqueNamedChunks([]).should.eql([]);
+			});
+		});
+		describe("given an array of distrinct variables", function() {
+			it("returns an array containing an array containing the variables", function() {
+				normalModule.splitVariablesInUniqueNamedChunks(variables).should.eql([variables]);
+			});
+		});
+		describe("given an array with duplicate variables", function() {
+			it("returns several arrays each containing only distinct variable names", function() {
+				normalModule.splitVariablesInUniqueNamedChunks(variables.concat(variables)).should.eql([variables, variables]);
+			});
+			describe("and a duplicate as the last variable", function() {
+				it("returns correctly split distinct arrays", function() {
+					normalModule.splitVariablesInUniqueNamedChunks(variables.concat(variables).concat(variables[0])).should.eql([variables, variables, [variables[0]]]);
+				});
+			});
+		});
+	});
+
+	describe("#applyNoParseRule", function() {
+		let rule;
+		let content;
+		describe("given a string as rule", function() {
+			beforeEach(function() {
+				rule = "some-rule";
+			});
+			describe("and the content starting with the string specified in rule", function() {
+				beforeEach(function() {
+					content = rule + "some-content";
+				});
+				it("returns true", function() {
+					normalModule.preventParsing(rule, content).should.eql(true);
+				});
+			});
+			describe("and the content does not start with the string specified in rule", function() {
+				beforeEach(function() {
+					content = "some-content";
+				});
+				it("returns false", function() {
+					normalModule.preventParsing(rule, content).should.eql(false);
+				});
+			});
+		});
+		describe("given a regex as rule", function() {
+			beforeEach(function() {
+				rule = /some-rule/;
+			});
+			describe("and the content matches the rule", function() {
+				beforeEach(function() {
+					content = rule + "some-content";
+				});
+				it("returns true", function() {
+					normalModule.preventParsing(rule, content).should.eql(true);
+				});
+			});
+			describe("and the content does not match the rule", function() {
+				beforeEach(function() {
+					content = "some-content";
+				});
+				it("returns false", function() {
+					normalModule.preventParsing(rule, content).should.eql(false);
+				});
+			});
+		});
+	});
+
+	describe("#preventParsing", function() {
+		let applyNoParseRuleSpy;
+		beforeEach(function() {
+			applyNoParseRuleSpy = sinon.stub();
+			normalModule.applyNoParseRule = applyNoParseRuleSpy;
+		});
+		describe("given no noParseRule", function() {
+			it("returns false", function() {
+				normalModule.preventParsing().should.eql(false);
+				applyNoParseRuleSpy.callCount.should.eql(0);
+			});
+		});
+		describe("given a noParseRule", function() {
+			let returnValOfSpy;
+			beforeEach(function() {
+				returnValOfSpy = Math.random() >= 0.5 ? true : false;
```

Comment:
i kind of wanted to make them be random, but i guess that doesnt help should one ever try to debug such a test. will change